Godeps/_workspace/src/github.com/ipfs/go-datastore/measure/measure.go

```diff
@@ -179,10 +179,9 @@ func (m *measure) Batch() (datastore.Batch, error) {
 func (mt *measuredBatch) Put(key datastore.Key, val interface{}) error {
 	mt.puts++
 	valb, ok := val.([]byte)
-	if !ok {
-		return datastore.ErrInvalidType
+	if ok {
+		_ = mt.m.putSize.RecordValue(int64(len(valb)))
```

Comment:
Here youre starting to touch on a change i've wanted to make to the datastore for a while now.

Instead of just throwing the length away if the value isnt bytes, how about we check if the value has a `Len()` method?

We could define an interface (hrm, this sounds familiar) that has something like:
```
type Value interface {
  Data() []byte
  Len() int
}
```
and then just type check for that object, and use its `Len()` method here if available.

The awesome thing this would let us do is pass the `Block` object from the blockstore *directly* into the datastore.Put. Once we get that working, we can extend the interface differently down the road (example: maybe something like a `WriteTo` method that would help us avoid keeping the blocks in memory)